dll/win32/syssetup/wizard.c

```diff
@@ -390,89 +391,217 @@ AckPageDlgProc(HWND hwndDlg,
+        error = RegSetValueExW(hKey, s_szCurrentVersion, 0, REG_SZ, (const BYTE *)L"5.2", cbData);
+        if (error)
+            break;
+
+        /* write WindowsNT CurrentBuildNumber */
+        cbData = sizeof(L"3790");
+        error = RegSetValueExW(hKey, s_szCurrentBuildNumber, 0, REG_SZ, (const BYTE *)L"3790", cbData);
+        if (error)
+            break;
+        
+        RegCloseKey(hKey);
+
+        /* open Control Windows key */
+        error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szControlWindows, 0, KEY_WRITE, &hKey);
+        if (error)
+            break;
+
+        /* write Control Windows CSDVersion */
+        dwValue = 0x200;
+        cbData = sizeof(dwValue);
+        error = RegSetValueExW(hKey, L"CSDVersion", 0, REG_DWORD, (const BYTE *)&dwValue, cbData);
+        if (error)
+            break;
+    } while (0);
+
+    if (hKey)
+        RegCloseKey(hKey);
 
     return error == ERROR_SUCCESS;
 }
 
-static void
-OnChooseServer(HWND hwndDlg)
+static BOOL
+WriteProductOptionWorkstation(void)
 {
-    WCHAR szText[256];
 
-    SetDlgItemTextW(hwndDlg, IDC_PRODUCT_SUITE, L"Terminal Server");
-    SetDlgItemTextW(hwndDlg, IDC_PRODUCT_TYPE, L"ServerNT");
+    HKEY hKey;
+    LONG error;
+    LPCWSTR pData;
+    DWORD dwValue, cbData;
+
+    do
+    {
+        /* open ProductOptions key */
+        error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szProductOptions, 0, KEY_WRITE, &hKey);
+        if (error)
+            break;
+
+        /* write ProductSuite */
+        pData = L"\0";
+        cbData = sizeof(L"\0");
+        error = RegSetValueExW(hKey, L"ProductSuite", 0, REG_MULTI_SZ, (const BYTE *)pData, cbData);
+        if (error)
+            break;
+
+        /* write ProductType */
+        pData = L"WinNT";
+        cbData = sizeof(L"WinNT");
+        error = RegSetValueExW(hKey, L"ProductType", 0, REG_SZ, (const BYTE *)pData, cbData);
+        if (error)
+            break;
+
+        RegCloseKey(hKey);
+
+        /* open ReactOS version key */
+        error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szRosVersion, 0, KEY_WRITE, &hKey);
+        if (error)
+            break;
+
+        /* write ReportAsWorkstation */
+        dwValue = 1;
+        cbData = sizeof(dwValue);
+        error = RegSetValueExW(hKey, L"ReportAsWorkstation", 0, REG_DWORD, (const BYTE *)&dwValue, cbData);
+        if (error)
+            break;
+
+        RegCloseKey(hKey);
+
+        /* open WindowsNT key */
+        error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szWindowsNT, 0, KEY_WRITE, &hKey);
+        if (error)
+            break;
+
+        /* write WindowsNT CurrentVersion */
+        cbData = sizeof(L"5.1");
+        error = RegSetValueExW(hKey, s_szCurrentVersion, 0, REG_SZ, (const BYTE *)L"5.1", cbData);
+        if (error)
+            break;
+
+        /* write WindowsNT CurrentBuildNumber */
+        cbData = sizeof(L"2000");
+        error = RegSetValueExW(hKey, s_szCurrentBuildNumber, 0, REG_SZ, (const BYTE *)L"2000", cbData);
```

Comment:
XPsp3 reports `2600`.
Is `2000` the actual value on S03sp2?

Reply:
You are correct concerning 2600:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/appcompat/shims/layer/versionlie.c;hb=98c17d3120b66c6c9507474cb5c3db1ca7b36ed3#l37

Reply:
![W2K3_SP2_Version](https://user-images.githubusercontent.com/32620577/81506150-f083dc80-92b9-11ea-845c-ee8b0f714ab1.png)

See Above, It should be 3790.

Reply:
@Doug-Lyons : the comment was about windows XP, not about windows 2003.

Reply:
Sorry, I saw the question from @SergeGautherie about S03sp2 and I was trying to clarify it.

Reply:
Doug, this line is about S03sp2 reporting as workstation.
Is that what you tested?

Reply:
Changed to `2600`.